float64[] position # x y, z
